craft_store/endpoints.py

```diff
@@ -0,0 +1,75 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Endpoint definitions for different services."""
+
+import dataclasses
+from typing import Any, Dict, Final, Sequence
+
+
+@dataclasses.dataclass(repr=True)
+class Endpoints:
+    """Endpoints used to make requests to a store.
+
+    :param whoami: path to the whoami API.
+    :param tokens: path to the tokens API.
+    :param tokens_exchange: path to the tokens_exchange API.
+    """
+
+    whoami: str
+    tokens: str
+    tokens_exchange: str
+
+    @staticmethod
+    def _get_token_request(
```

Comment:
It will be used somewhat like this from the StoreClient class:

```python
def login(self, permissions: List[str], description: str, ttl: str) -> None:
        token_request = self._endpoints._get_token_request(
            permissions=permissions, description=description, ttl=ttl
        )

        token = self._get_token(token_request)
        # ...
        # ...
```

So not exposed to the user of the library in any way.